tinygrad/helpers.py

```diff
@@ -89,6 +97,8 @@ def from_np(x) -> DType: return asdict(dtypes())[np.dtype(x).name]
   uint32: Final[DType] = DType(1, 4, "uint", np.uint32)
   uint64: Final[DType] = DType(2, 8, "uint64", np.uint64)
 
+# HACK: staticmethods are not callable in 3.8 so we have to compare the class
+DTYPES_DICT = {k: v for k, v in dtypes.__dict__.items() if not k.startswith('__') and not callable(v) and not v.__class__ == staticmethod}
```

Comment:
You could also remove the `dataclass` decorator, instantiate `dtypes()` once at the end of this file and cache the `from_np` function. This will also cache the repeated `np.dtype(x).name` calls.